awx/ui_next/src/screens/Host/Host.jsx

```diff
@@ -23,8 +26,16 @@ class Host extends Component {
       hasContentLoading: true,
       contentError: null,
       isInitialized: false,
+      toggleLoading: false,
+      toggleError: null,
+      deletionError: false,
+      isDeleteModalOpen: false,
     };
     this.loadHost = this.loadHost.bind(this);
+    this.handleHostToggle = this.handleHostToggle.bind(this);
+    this.handleToggleError = this.handleToggleError.bind(this);
+    this.handleHostDelete = this.handleHostDelete.bind(this);
+    this.toggleDeleteModal = this.toggleDeleteModal.bind(this);
```

Comment:
I chose to keep of the delete modal toggling and delete functionality since this component already had state.  I was also trying to keep HostDetails to be stateless

Reply:
In my opinion, these methods belong within the context of HostDetail.jsx. Host.jsx acts primarily as a "route" file for the subtab views. Since only the HostDetail view cares about these methods, it would make more sense for the delete and toggle logic to live in there. This would also simplify the Host.jsx file and reduce the number of props drilled down.

Reply:
Just want to make sure...does this follow the UX of other error/delete on detail views.  what @marshmalien is making sense of where it should live, but I also want to make sure we aren't doing something "special" here that we're not doing in other places that have the same concepts.

Reply:
I was able to move most of this work into `HostDetail` file which reduced the props being passed in. I believe that `Host` is now ready to be refactored into a functional component but that can be done during another PR.

Reply:
@jlmitch5 
> Just want to make sure...does this follow the UX of other error/delete on detail views. what @marshmalien is making sense of where it should live, but I also want to make sure we aren't doing something "special" here that we're not doing in other places that have the same concepts.

JobDetail (handles deletion errors as well as delete), InventoryDetail (handles delete but does not handle deletion errors), InventoryGroupDetail (handle delete and deletion errors)

OrganizationDetail, UserDetail JobTemplateDetail have no delete functionality at all, and probably should follow the same pattern as the files mentioned above.